sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -354,6 +355,13 @@ class ParquetFileFormat
         }
       } else {
         logDebug(s"Falling back to parquet-mr")
+
+        if (SQLConf.get.parquetFieldIdEnabled &&
+            ParquetUtils.hasFieldIds(requiredSchema)) {
+          throw new IOException("Parquet-mr reader does not support schema with field IDs." +
+            s" Please choose a different Parquet reader. Read schema: ${requiredSchema.json}")
```

Comment:
Maybe we should recommend disabling the flag instead or in addition to?

Reply:
sure, sounds good, how about "Please choose a different Parquet reader or disable reading field id by setting `spark.sql.parquet.fieldId.enabled = false`"

Reply:
Yes, that could work. I would probably update your statement to something like this:
> Parquet-mr reader does not currently support reading schema with field IDs. Please enable vectorised Parquet reader or set <config> to false.

Reply:
deprecated as parquet-mr is supported now.